### ISSUE TITLE
Removal of 2020-resolver enforcing breaks non developer installations

### ISSUE THREAD

**Issue by dliappis** [elastic/rally#1184]
As described in https://github.com/elastic/rally/pull/1183#issuecomment-779094529, the removal of explicitly using the `2020-resolver` of pip in #1151 breaks installations of the latest [2.0.4 Rally release](https://github.com/elastic/rally/releases/tag/2.0.4) for several people who haven't upgraded their pip.

Note: We are requiring pip to be upgraded in [here](https://esrally.readthedocs.io/en/stable/install.html#python) but this command won't help for venv installations.

## Reproduction

<details><summary>Reproduction with Python 3.8.0</summary>

```
$ docker run --rm -ti python:3.8.0 /bin/bash

root@6c5413504db2:/# python3 --version
Python 3.8.0

root@6c5413504db2:/# python3 -mpip --version
pip 19.3.1 from /usr/local/lib/python3.8/site-packages/pip (python 3.8)

root@6c5413504db2:/# python3 -m pip install --user --upgrade pip
Collecting pip
  Downloading https://files.pythonhosted.org/packages/fe/ef/60d7ba03b5c442309ef42e7d69959f73aacccd0d86008362a681c4698e83/pip-21.0.1-py3-none-any.whl (1.5MB)
     |████████████████████████████████| 1.5MB 850kB/s 
Installing collected packages: pip
  WARNING: The scripts pip, pip3 and pip3.8 are installed in '/root/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
Successfully installed pip-21.0.1
WARNING: You are using pip version 19.3.1; however, version 21.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.

root@6c5413504db2:/# python3 -mpip --version
pip 21.0.1 from /root/.local/lib/python3.8/site-packages/pip (python 3.8)

root@6c5413504db2:/# python3 -m venv .venv

root@6c5413504db2:/# source .venv/bin/activate

(.venv) root@6c5413504db2:/# python3 -m pip install esrally
Collecting esrally

... several output lines snipped ...

ERROR: botocore 1.13.50 has requirement urllib3<1.26,>=1.20; python_version >= "3.4", but you'll have urllib3 1.26.3 which is incompatible.
Installing collected packages: urllib3, docutils, six, python-dateutil, jmespath, botocore, s3transfer, boto3, pycparser, cffi, google-crc32c, google-resumable-media, tabulate, certifi, ijson, idna, multidict, yarl, attrs, async-timeout, typing-extensions, chardet, aiohttp, elasticsearch, MarkupSafe, Jinja2, zipp, importlib-metadata, pyrsistent, jsonschema, psutil, yappi, py-cpuinfo, thespian, pyasn1, pyasn1-modules, cachetools, rsa, google-auth, esrally
  Running setup.py install for ijson ... done
  Running setup.py install for pyrsistent ... done
  Running setup.py install for yappi ... done
  Running setup.py install for py-cpuinfo ... done
  Running setup.py install for thespian ... done
Successfully installed Jinja2-2.10.3 MarkupSafe-1.1.1 aiohttp-3.7.3 async-timeout-3.0.1 attrs-20.3.0 boto3-1.10.32 botocore-1.13.50 cachetools-4.2.1 certifi-2020.12.5 cffi-1.14.5 chardet-3.0.4 docutils-0.15.2 elasticsearch-7.9.1 esrally-2.0.4 google-auth-1.22.1 google-crc32c-1.1.2 google-resumable-media-1.1.0 idna-3.1 ijson-2.6.1 importlib-metadata-3.4.0 jmespath-0.10.0 jsonschema-3.1.1 multidict-5.1.0 psutil-5.8.0 py-cpuinfo-7.0.0 pyasn1-0.4.8 pyasn1-modules-0.2.8 pycparser-2.20 pyrsistent-0.17.3 python-dateutil-2.8.1 rsa-4.7 s3transfer-0.2.1 six-1.15.0 tabulate-0.8.7 thespian-3.10.1 typing-extensions-3.7.4.3 urllib3-1.26.3 yappi-1.2.3 yarl-1.6.3 zipp-3.4.0
WARNING: You are using pip version 19.2.3, however version 21.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
```
</details>

## Fixes or workarounds

1. Address in the docs [here](https://esrally.readthedocs.io/en/stable/install.html#installing-rally) and [here](https://esrally.readthedocs.io/en/stable/install.html#virtualenv-install) simply instructing users to `python3 -mpip install --upgrade pip` before installing Rally. This will be in addition to the [prerequisites](https://esrally.readthedocs.io/en/stable/install.html#python) explicitly mentioning the upgrade command but is definitely required for the `venv` section, as described in the reproduction above.
2. Similar to 1. but instead use `export PIP_USE_FEATURE=2020-resolver`
3. Require a higher default version than 3.8.0 ([here](https://github.com/elastic/rally/blob/b8592a6071e549be99ac5538f960fe3026e513fb/.ci/variables.json#L3) and in [docs](https://github.com/elastic/rally/blob/master/docs/install.rst#python)) as newer versions of Python come with a newer pip version -- since 20.3 the 2020-resolver is on by default: https://discuss.python.org/t/announcement-pip-20-3-release/5948
4. Something else?

**Comment by dliappis** [comment]
Had an offline discussion with @danielmitterdorfer and given the seriousness for the entry level experience, we'll use Fix 1 and try to also address the stable docs to reflect the updated instructions until we've published a new version.

I've raised #1151 to address docs, but we won't close this ticket until we've also added an IT test allowing us to detect such a problem from happening again in the future.

**Comment by dliappis** [comment]
Stable docs now include the docs changes brought in #1185 ; this is done with a new [stable branch](https://github.com/elastic/rally/tree/stable) which contains everything that the [2.0.4 tag](https://github.com/elastic/rally/releases/tag/2.0.4) includes plus the specific docs commit.

One we've released a newer Rally version, we'll delete the stable branch.

**Comment by dliappis** [comment]
Closing this issue due to the fix in #1185. Rally 2.1.0 will include the docs updates and we'll delete the `stable` branch.

**Comment by dliappis** [comment]
Rally 2.1.0 is [out](https://pypi.org/project/esrally/), and the `stable` branch has been deleted.
